ledger-core/application/testwalletapi/statemachine/statemachine_test.go

```diff
@@ -49,42 +53,114 @@ func TestSMTestAPICall_Migrate(t *testing.T) {
 	slotMachine.Start()
 	defer slotMachine.Stop()
 
-	slotMachine.MessageSender.SendRole.Set(func(_ context.Context, msg payload.Marshaler, role node.DynamicRole, object reference.Global, pn pulse.Number, _ ...messagesender.SendOption) error {
-		res := msg.(*payload.VCallRequest)
-		assert.Equal(t, request.Callee, res.Callee)
-		assert.Equal(t, node.DynamicRoleVirtualExecutor, role)
-		assert.Equal(t, request.Callee, object)
-		return nil
-	})
-
 	smRequest := SMTestAPICall{
 		requestPayload: request,
 	}
 
 	smWrapper := slotMachine.AddStateMachine(ctx, &smRequest)
 
-	slotMachine.RunTil(smWrapper.BeforeStep(smRequest.stepSendRequest))
+	slotMachine.RunTil(smWrapper.BeforeStep(smRequest.stepRegisterBargeIn))
+
+	require.NotEqual(t, APICaller, smRequest.requestPayload.Caller)
+	require.True(t, smRequest.registeredBargeInRef.IsEmpty())
+	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepRegisterBargeIn))
 
 	outgoingCall := smRequest.requestPayload.CallOutgoing
 
 	outgoingRef := reference.NewRecordOf(APICaller, outgoingCall)
 	_, bargeIn := slotMachine.SlotMachine.GetPublishedGlobalAliasAndBargeIn(outgoingRef)
-	assert.NotNil(t, bargeIn)
+	require.NotNil(t, bargeIn)
+
+	require.Equal(t, APICaller, smRequest.requestPayload.Caller)
+	require.NotEmpty(t, smRequest.requestPayload.CallOutgoing)
+	require.Equal(t, outgoingRef, smRequest.registeredBargeInRef)
+
+	{
+		slotMachine.Migrate()
+		slotMachine.RunTil(smWrapper.AfterMigrate(smRequest.migrateBeforeSendRequest))
+		// check that we remove bargein after migrate
+		_, bargeIn = slotMachine.SlotMachine.GetPublishedGlobalAliasAndBargeIn(outgoingRef)
+		require.Nil(t, bargeIn)
+	}
+
+	slotMachine.RunTil(smWrapper.AfterStep(smRequest.stepRegisterBargeIn))
+	newOutgoingCall := smRequest.requestPayload.CallOutgoing
+	var newBargeIn smachine.BargeInHolder
+	{
+		newOutgoingRef := reference.NewRecordOf(APICaller, newOutgoingCall)
```

Comment:
can you add one check: `outgoingRef` != `newOutgoingRef`?

Reply:
There is, little bit lower 
require.NotEqual(t, outgoingCall, newOutgoingCall)

Reply:
no, it's direct check

Reply:
done